### PR TITLE
add body to buildOptions and add report endpoint reason

### PR DESCRIPTION
**Pull request by Timgur**
just want to make sure that the newest report endpoint is added and working before submitting this.

### CODE REVIEW

File: build/imgur.js [hunk 1]
```diff
@@ -8,20 +8,21 @@
...
+            var body = arguments[3] === undefined ? {} : arguments[3];
+
+            return { apiUrl: apiUrl, path: path, method: method, body: body };
         },
         bearer: ''
     };
 
     var imgurAPICall = function imgurAPICall(options) {
-        ['method', 'apiUrl', 'path'].forEach(function (option) {
+        ['method', 'apiUrl', 'path', 'body'].forEach(function (option) {
```
Comment (Polyrhythm) [inline build/imgur.js]: prefer `(option) => {}` syntax for anonymous functions inside iterators

File: build/imgur.js [hunk 2]
```diff
@@ -101,9 +102,9 @@
 
             return this.imgurAPICall(options);
         },
-        report: function report(hash) {
+        report: function report(hash, reason) {
             var path = '' + this.path + '/' + hash + '/report';
-            var options = utils.buildOptions(this.apiUrl, path, 'post');
+            var options = utils.buildOptions(this.apiUrl, path, 'post', { reason: reason });
```
Comment (Polyrhythm) [inline build/imgur.js]: `{ reason }`

File: build/imgur.js [hunk 3]
```diff
@@ -151,9 +152,9 @@
 
             return this.imgurAPICall(options);
         },
-        report: function report(commentId) {
+        report: function report(commentId, reason) {
             var path = '' + this.path + '/' + commentId + '/report';
-            var options = utils.buildOptions(this.apiUrl, path, 'post');
+            var options = utils.buildOptions(this.apiUrl, path, 'post', { reason: reason });
```
Comment (Polyrhythm) [inline build/imgur.js]: `{ reason }`

File: build/imgur.js [hunk 4]
```diff
@@ -8,20 +8,21 @@
         API_VERSION: '3',
         CLIENT_ID: '',
         buildOptions: function buildOptions(apiUrl, path, method) {
-            return { apiUrl: apiUrl, path: path, method: method };
+            var body = arguments[3] === undefined ? {} : arguments[3];
+
+            return { apiUrl: apiUrl, path: path, method: method, body: body };
```
Comment (Polyrhythm) [inline build/imgur.js]: `return { apiUrl, path, method, body };`

File: build/imgur.js [hunk 5]
```diff
@@ -8,20 +8,21 @@
         API_VERSION: '3',
         CLIENT_ID: '',
         buildOptions: function buildOptions(apiUrl, path, method) {
-            return { apiUrl: apiUrl, path: path, method: method };
+            var body = arguments[3] === undefined ? {} : arguments[3];
```
Comment (Polyrhythm) [inline build/imgur.js]: Would it be possible to provide default values and then send a reasonable body other than `undefined`? Not sure what the reason for adding a `body` is or what it does.

Comment (Polyrhythm) [inline build/imgur.js]: Prefer `let` or `const` over `var`

File: build/imgur.js (hunk 2 above)
Comment (Polyrhythm) [inline build/imgur.js]: prefer `let` or `const` over `var`

File: build/imgur.js (hunk 1 above)
Comment (Timgur) [inline build/imgur.js]: this is the compiled js

File: build/imgur.js (hunk 2 above)
Comment (Timgur) [inline build/imgur.js]: this is the compiled js

File: build/imgur.js (hunk 5 above)
Comment (Timgur) [inline build/imgur.js]: compile build

File: build/imgur.js (hunk 3 above)
Comment (Timgur) [inline build/imgur.js]: compile js

File: build/imgur.js (hunk 4 above)
Comment (Timgur) [inline build/imgur.js]: src looks like:
screen shot 2015-07-17 at 8 50 03 am

**Pull request merged**